### PR TITLE
Add "contributing to Nixpkgs" tutorial

### PR DESCRIPTION
**Pull request by proofconstruction**
While writing *Packaging Existing Software with Nix*, I originally had a section at the end where the reader is walked through preparing the package for submission to Nixpkgs. This is a cleaned-up version of that which maintains continuity with the other tutorial, and should help get people contributing faster.

There is a lot more to be said on this topic, but probably all of *that* belongs in a recipe instead.

### CODE REVIEW

File: source/tutorials/contributing-to-nixpkgs.md
Comment (zmitchell) [inline source/tutorials/contributing-to-nixpkgs.md:182]: You should try running the package first :)

**Review by zmitchell — CHANGES_REQUESTED** [review]
Generally I think you can cut down how much of this is rehashing the previous tutorial, but other than that I'm a fan of the content.

I think this belongs in the learning journey section given that it's a direct follow up to the learning journey tutorial it refers to.

File: source/tutorials/contributing-to-nixpkgs.md
Comment (yukiisbored) [reply source/tutorials/contributing-to-nixpkgs.md:15]: Unlike links, this uses the `packaging-existing-software` reference and will work no matter where it is placed in the file system as long as the reference exists somewhere.

File: source/tutorials/contributing-to-nixpkgs.md
Comment (infinisil) [inline source/tutorials/contributing-to-nixpkgs.md:139]: You should use [`pkgs/by-name`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#readme) for new packages now! No need to edit `all-packages.nix` either that way :)

File: source/tutorials/contributing-to-nixpkgs.md
Comment (infinisil) [inline source/tutorials/contributing-to-nixpkgs.md:190]: This is a better docs link: https://github.com/NixOS/nixpkgs/tree/master/maintainers#readme

**Review by infinisil — CHANGES_REQUESTED** [review]

**Review by bzm3r — COMMENTED** [review]
I like it! I learned just from reading the PR, so yeah, it would be really nice to have this on nix.dev!

I think the changes suggested by @infinisil and @zmitchell are quite useful. So, once those get merged....

File: source/tutorials/contributing-to-nixpkgs.md
Comment (bzm3r) [reply source/tutorials/contributing-to-nixpkgs.md:25]: Yes! I strongly agree with this.

**Comment by infinisil** [comment]
@bzm3r Since the original author seems busy, could you create a new PR taking the feedback into account?

**Comment by proofconstruction** [comment]
Decided in [this comment](https://github.com/NixOS/nix.dev/pull/831#issuecomment-1879545180) to close this and merge anything useful into the `CONTRIBUTING.md` in nixpkgs, etc.

**Pull request closed without merging**
